Sources/PostgresNIO/Connection/PostgresConnection.swift

```diff
@@ -77,13 +77,23 @@ public final class PostgresConnection {
             ///
             /// - Default: 5432
             public var port: Int
-
-            public init(host: String, port: Int = 5432) {
+            
+            /// Require connection to provide `BackendKeyData`
```

Comment:
This issue may go beyond RDS Proxy to include any Postgres-compatible service that does not support canceling a query, so I will make a note of that general issue as well. (One of my colleagues noticed that CockroachDB used to have a similar issue with a Postgres library in Python until they added a workaround: https://github.com/cockroachdb/cockroach/pull/60281.)

I'm wondering if we'd be better off with something like `canCancelQueries` instead of `requireBackendKeyData`. The behavior wouldn't have to change, but the purpose would be clearer to anyone configuring their Postgres integration. The `BackendKeyData` functionality in Postgres is pretty obscure.

Reply:
The problem I have with `canCancelQueries` is that it would make users think they can not cancel queries in Swift. However this is not true. After a cancellation we can just ignore further data... or we just close the connection. Cancellation is not required to go trough the crazy Postgres way... However server cancellation works if we get backendKeyData.

So we could name this property `mustSupportServerSideCancellation`. Not sure if that makes anything clearer to users... 

I agree that this behavior goes beyond RDS proxy (for this reason the property shall not be called `supportRDSProxy`), but it is a use-case for it. For this reason I think it might make adopters life easier if they search for rds proxy in the code base.